src/snyk/common/languageServer/middleware.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  CancellationToken,
+  ConfigurationParams,
+  ConfigurationRequestHandlerSignature,
+  Middleware,
+  ResponseError,
+  WorkspaceMiddleware,
+} from '../vscode/types';
+import { ClientSettings, ServerSettings } from './settings';
+
+export type LanguageClientWorkspaceMiddleware = Partial<WorkspaceMiddleware> & {
+  configuration: (
+    params: ConfigurationParams,
+    token: CancellationToken,
+    next: ConfigurationRequestHandlerSignature,
+  ) => Promise<ResponseError<void> | ServerSettings[]>;
+};
+
+export class LanguageClientMiddleware implements Middleware {
+  workspace: LanguageClientWorkspaceMiddleware = {
```

Comment:
Why not an explicit class definition instead of a "lambda" or whatever they are called in VSCode?

Reply:
It's an object assignment here, passing workspace middleware. What would be the benefit of defining a separate class in this case?

Reply:
Easier to read and smaller units of compilation. But basically just a question of style-preference, so nothing that needs to change.